src/libappimage/libappimage.cpp

```diff
@@ -73,9 +73,9 @@ char** appimage_list_files(const char* path) {
 
         return result;
     } catch (const std::runtime_error& err) {
-        libappimageLogger.error() << " at " << __FUNCTION__ << " : " << err.what() << std::endl;
+       logger->log(LogLevel::ERROR, std::string(__FUNCTION__) + " : " + err.what());
```

Comment:
`__FUNCTION__` macro should be read automatically, no need to use it manually here.

Reply:
Not quite sure of what you mean. If you want it to be called from the 'log'  function, it will report `log` as the function name instead of the actual function.

Reply:
You could define a macro.

Reply:
To be honest, I'm not a big fan of macros. They tend to become obscure and are difficult to debug. 

Would you mind to check the new version of the code? It's much more legible.